testcases/kdump/lib/crasher/crasher.c

```diff
@@ -82,28 +81,24 @@ static int crasher_write(struct file *file, const char *buffer,
 				   somtimes white lies are ok */
 }
 
+struct file_operations proc_fops = {
+read: crasher_read,
+write: crasher_write
+};
+
 /* create a directory in /proc and a debug file in the new directory */
 
 int crasher_init(void)
 {
-	struct proc_dir_entry *crasher_proc;
-
-	printk("loaded crasher module\n");
-
-	/* build a crasher file that can be set */
-	if ((crasher_proc = create_proc_entry(CRASH, 0, NULL)) == NULL) {
-		return -ENOMEM;
-	}
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 30)
-	crasher_proc->owner = THIS_MODULE
-#endif
-	crasher_proc->read_proc = crasher_read;
-	crasher_proc->write_proc = crasher_write;
-	return 0;
+printk("loaded crasher module\n");
+proc_create(CRASH,0,NULL,&proc_fops);
+return 0;
 }
```

Comment:
You should do this in a backward compatible manner. I.e. Keep the ifdefs and add a few more as well.

Moreover the changes messed up the coding style. You should use checkpatch.pl (shipped with Linux kernel sources) to check your patches before the submission.